addons/mrp/models/stock_move.py

```diff
@@ -309,3 +338,16 @@ def _compute_kit_quantities(self, product_id, kit_qty, kit_bom, filters):
             return min(qty_ratios) // 1
         else:
             return 0.0
+
+    def _decrease_initial_demand(self, qty, stream='UP'):
+        done_move_to_return = []
+        finished_moves = self.env['stock.move']
+        for move in self:
+            if move.production_id and move.state not in ('done, cancel'):
+                try:
+                    if move.production_id.state == 'confirmed':
+                        move.production_id._update_quantity(move.production_id.product_qty - qty)
+                        finished_moves |= move
+                except UserError:
```

Comment:
when is the user error raised?

Reply:
it's this one https://github.com/odoo/odoo/pull/43192/commits/c2385b1444c3db9bf0ddb3ecd0175f4773ff7dc2#diff-0bff01b88e2f2c3fa0962967170420deL51